README.md

```diff
@@ -0,0 +1,130 @@
+
+![Screenshot of real-life example of wrapping text with spans to highlight find-on-page results in Chromium Code Search](resources/custom-find-on-page-spans.png)
+
+However, implementing this can be somewhat cumbersome because we need to manipulate the DOM tree by inserting new objects into it. This can lead to many issues:
+
+- We should be careful not to break other processing of the DOM or cached elements, as the number of children/siblings of a node could change. This means the rest of the webpage should support dynamic changes in the DOM and act accordingly.
+- Some cases are not easy to handle. For example, highlighting a chunk of text that crosses a node boundary.
+- Performance could be impacted due to the DOM growing too much. This could affect operations such as rendering frames or performing other actions needed by the site itself. See [Demo 2](https://github.com/ffiori/highlight-api-demos/tree/add-readme#demo-2-performance-of-both-approaches) for a performance comparison with the Highlight API.
+- Reverting the operation is not easy in non-trivial sites. We should traverse the DOM looking for spans with this specific id or class and delete them while reinserting their content into the same place it was before. Furthermore, this operation could be computationally expensive if the DOM's got too big and there are many highlighted spans to remove.
+
+Here is a possible way of implementing this technique to highlight selected text that would only work for simple sites:
+
+```Javascript
+<style>
+  .highlighted-with-span {
+    color: red;
+  }
+</style>
+
+<script>
+  const highlightedWithSpanClass = "highlighted-with-span";
+
+  function highlightSelectionWithSpans() {
+    if (window.getSelection().isCollapsed)
+      return;
+    try {
+      let span = document.createElement('span');
+      span.className = highlightedWithSpanClass;
+      window.getSelection().getRangeAt(0).surroundContents(span);
+    }
+    catch (error) {
+      ; // Treat error
+    }
+  }
+
+  function clearSpans() {
+    let spans = document.getElementsByClassName(highlightedWithSpanClass);
+    while(spans.length) {
+      let span = spans[0];
+      span.parentNode.insertBefore(span.firstChild, span);
+      span.parentNode.removeChild(span);
+    }
+  }
+<script>
+```
+
+## Examples
+
+This section describes two examples of sites that make use of the Highlight API and intend to compare it to using spans to achieve the same goal. Note that, to be able to run the demos, Edge or Chrome >= 93 is needed (although it's recommended to use >= 95 for testing including the latest patches), and the browser should run with the flag `--enable-experimental-web-platform-features` enabled (it can be turned on in [edge://flags](edge://flags/#enable-experimental-web-platform-features) or [chrome://flags](chrome://flags/#enable-experimental-web-platform-features) under the name of 'Experimental Web Platform features').

```

Comment:
```suggestion
This section describes two demos that make use of the Highlight API and intend to compare it to using spans to achieve the same goal. Note that, to be able to run the demos, Edge or Chrome >= 93 is needed (although it's recommended to use >= 95 for testing including the latest patches), and the browser should run with the flag `--enable-experimental-web-platform-features` enabled (it can be turned on in [edge://flags](edge://flags/#enable-experimental-web-platform-features) or [chrome://flags](chrome://flags/#enable-experimental-web-platform-features) under the name of 'Experimental Web Platform features').

```

Reply:
(Suggesting this change because I think the current wording implies that these are live sites "in the wild" that are already using the API, rather than something that we've just created).

Reply:
Makes sense, thanks!